packages/frontend/src/providers/TokensProvider.tsx

```diff
@@ -37,36 +44,78 @@ const TokensProvider: FunctionComponent<{ children: ReactNode }> = ({ children }
     { enabled: false }
   );
 
-  const fetchTokens = async () => {
-    const { data: tokens } = await refetch();
+  const { refetch: refetchToTokens } = useQuery(
+    ['tokens'],
+    async () => {
+      const data = await sifi.getTokens(toChain.id);
+
+      return getOrderedTokenList(data);
+    },
+    { enabled: false }
+  );
 
-    if (tokens) {
-      setTokens(tokens);
+  const fetchTokens = async (
+    chain: Chain,
+    setTokens: React.Dispatch<React.SetStateAction<Token[]>>,
```

Comment:
Passing state setters + fetch function as callbacks here it too much I think.
Function is doing a lot more than fetching tokens as the name would suggest.

Fine for now but we should refactor this.

Reply:
ngl this was GPT-4

Addressed in 80f0dce4f4b4e5252e84695808653f0df85daa4f.